src/Main.elm

```diff
@@ -2,72 +2,137 @@ module Main exposing (main)
 
 import Browser
 import Data
+import UserGroup exposing (UserGroup)
+import ContactDetails
+import Settings
+import Tags
 import Html exposing (Html)
 import Html.Attributes as Attrs
+import Json.Decode exposing (decodeString, errorToString)
 
 
 
 ---- MODEL ----
 
 
 type alias Model =
-    {}
+    Result Json.Decode.Error
+        { name : String
+        , contactDetails : ContactDetails.State
+        , settings : Settings.State
+        , tags : Tags.State
```

Comment:
NIT: I would probably use `Model` instead of `State`. Can you explain why you went with `State` here?

Reply:
When using the word "model" you are typically referring to a [data model](https://en.wikipedia.org/wiki/Data_model) in the sense of modelling real world data or business objects in terms of a data structure and semantics. While "state" is used to refer to a data structure that holds values representing the [internal state](https://en.wikipedia.org/wiki/State_(computer_science)) of your program at a given moment. I think it is an important distinction to make and keep in mind for conceptual and architectural reasons.

So in this case the `UserGroup` type is a model (it is a record describing the real world concept of a user group), while the `ContactDetails.State` record is a state (describing the input and validation state of the form/UI). In this case the initial state is derived from the model, but they are not the same thing.